### PR TITLE
Update howto for finetuning on MPS

### PR DESCRIPTION
**Pull request by carmocca**
Addressing https://github.com/Lightning-AI/lit-parrot/pull/94#issuecomment-1567364223

### CODE REVIEW

File: finetune_adapter.py [hunk 1]
```diff
@@ -64,7 +65,10 @@ def main(
 
     config = Config.from_name(name=checkpoint_dir.name, block_size=max_seq_length)
 
-    with EmptyInitOnDevice(device=fabric.device, dtype=torch.float32 if fabric._precision.precision == "32-true" else torch.bfloat16):
+    with EmptyInitOnDevice(
+        device=fabric.device,
+        dtype=torch.float32 if fabric.strategy.precision.precision == "32-true" else torch.bfloat16
```
Comment (carmocca) [inline finetune_adapter.py:70]: Fabric doesn't offer any mapping from precision str/class to torch.dtype. So any other values would need modification here

File: howto/finetune_adapter.md [hunk 2]
```diff
@@ -43,6 +43,8 @@ This script will save checkpoints periodically to the `out_dir` directory. If yo
 python finetune_adapter.py --out_dir out/adapter/my-model-finetuned
 ```
 
+If your GPU does not support `bfloat16` (for instance, on MPS), you can pass the `--precision 32-true` argument.
```
Comment (lantiga) [inline howto/finetune_adapter.md:46]:
```suggestion
If your GPU does not support `bfloat16`, you can pass the `--precision 32-true` argument.

For instance, to fine-tune on MPS (the GPU on Macs), you can run

```bash
python finetune_adapter.py --out_dir out/adapter/my-model-finetuned --device mps --precision 32-true
```

Note that `mps` will be picked up automatically by the default `device="auto"`, it's spelled out for clarity here.
```

Comment (lantiga) [reply howto/finetune_adapter.md:46]: please see the raw comment ^, GitHub is messing with the formatting

**Review by lantiga — APPROVED** [review]

File: finetune_adapter.py (hunk 1 above)
Comment (awaelchli) [reply finetune_adapter.py:68]: One could also just use the local variable `precision` here.

**Review by lantiga — APPROVED** [review]

File: howto/finetune_adapter.md [hunk 3]
```diff
@@ -58,6 +58,15 @@ or for Adapter V2
...
 ```
 
+If your GPU does not support `bfloat16`, you can pass the `--precision 32-true` argument.
+For instance, to fine-tune on MPS (the GPU on modern Macs), you can run
+
+```bash
+python finetune_adapter.py --out_dir out/adapter/my-model-finetuned --precision 32-true
+```
+
+Note that `mps` will be picked up automatically by the default `device="auto"`, it's spelled out for clarity here.
```
Comment (lantiga) [inline howto/finetune_adapter.md:68]:
```suggestion
Note that `mps` as the accelerator will be picked up automatically by Fabric when running on a modern Mac.
```

**Pull request merged**